treeherder/perf/secretary_tool.py

```diff
@@ -0,0 +1,81 @@
+import logging
+from datetime import (datetime,
+                      timedelta)
+
+import simplejson as json
+from django.conf import settings as django_settings
+
+from treeherder.perf.models import (BackfillRecord,
+                                    PerformanceSettings)
+
+logger = logging.getLogger(__name__)
+
+# TODO: extract strings into constants
+TIME_TO_MATURE = timedelta(hours=4)
+
+
+class SecretaryTool:
+    """
+    Tool used for doing the secretary work in the Performance Sheriff Bot.
+    """
+
+    def __init__(self):
+        pass
+
+    @classmethod
+    def manage_settings(cls):
+        perf_sheriff_settings, created = PerformanceSettings.objects.get_or_create(
+            name="perf_sheriff_bot",
+            defaults={"settings": cls._get_default_settings()},
+        )
+
+        if created:
+            logger.info("Performance settings for perf_sheriff_bot not found. Creating with defaults.")
+
+        # reset limits if the settings expired
+        settings = json.loads(perf_sheriff_settings.settings)
+        if cls.are_expired(settings):
+            logger.info(f"Settings are expired. Expired settings: {settings}")
+            # self.log.warning(f"Failed to compute report for alert summary {summary}. {ex}")
+
+            perf_sheriff_settings.settings = cls._get_default_settings()
+            perf_sheriff_settings.save()
+
+    @classmethod
+    def prepare_for_processing(cls):
+        # get all PRELIMINARY records and set them to READY_FOR_PROCESSING
+        preliminary_records = BackfillRecord.objects.filter(status=BackfillRecord.PRELIMINARY)
```

Comment:
The reporter tool generates reports *or* deletes them starting from the `BackfillReport`.
Just for flow consistency, I think we can follow that approach here.

Basically, identify all `BackfillReport`s which aren't yet frozen & are mature. Iterate through their PRELIMINARY children, so they're marked as READY_FOR_PROCESSING.
Then set each of these reports as frozen.

@airimovici what do you think?